uaclient/util.py

```diff
@@ -61,6 +62,31 @@ def encode_text(text, encoding='utf-8'):
     return text.encode(encoding)
 
 
+def get_dict_deltas(orig_dict, new_dict, path=''):
+    """Return a dictionary of delta between orig_dict and new_dict."""
+    deltas = {}
+    for key, value in orig_dict.items():
+        new_value = new_dict.get(key, DROPPED_DICT_KEY)
+        if isinstance(value, dict):
+            if path:
+                sub_path = path + '.' + key
+            else:
+                sub_path = key
+            sub_delta = get_dict_deltas(
+                value, new_dict.get(key, {}), path=sub_path)
+            if sub_delta:
+                deltas[key] = sub_delta
+        elif value != new_value:
+            key_path = key if not path else path + '.' + key
```

Comment:
This is the same logic as `sub_path` in the other branch of this if; can we refactor it out so we only calculate it once?

Reply:
refactored logic to calculate key_path in one place.